chalice/awsclient.py

```diff
@@ -121,15 +132,55 @@ def update_function(self, function_name, zip_contents,
             # We're going to need this moving forward anyways,
             # more config options besides env vars will be added.
             'Environment': {'Variables': environment_variables},
-        }
+        }  # type: Dict[str, Any]
         if runtime is not None:
             kwargs['Runtime'] = runtime
+        if timeout is not None:
+            kwargs['Timeout'] = timeout
+        if memory_size is not None:
+            kwargs['MemorySize'] = memory_size
         lambda_client.update_function_configuration(**kwargs)
         if tags is not None:
-            lambda_client.tag_resource(Resource=return_value['FunctionArn'],
-                                       Tags=tags)
+            function_arn = return_value['FunctionArn']
```

Comment:
The tag manipulation (figuring what exactly to tag) feels too level for this function.  Please move this out into a separate method, something like:

```python
if tags is not None:
    self._update_function_tags(return_value['FunctionArn'], tags)
```